app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -63,21 +68,27 @@ function RawScatterPlot({
     })
 
     const startTime = performance.now()
-
     Plotly.react(graphElementId, plotlyTraces, layout)
+
     sortLegend({
       graphElementId,
       isAnnotatedScatter,
       annotType: scatter.annotParams.type,
       genes: scatter.genes
     })
-
     perfTimes.plot = performance.now() - startTime
 
     logScatterPlot(
       { scatter, genes, width: dimensions.width, height: dimensions.height },
       perfTimes
     )
+
+    if (isCorrelatedScatter) {
+      // compute correlation stats, but do it after the graph render so it doesn't delay
+      // the visualizations or impact logging
+      const spearmanRho = new SpearmanRho(scatter.data.x, scatter.data.y)
+      spearmanRho.calc().then(value => setSpearman(value))
+    }
```

Comment:
Let's swap these -- the time needed to show this important statistic seems relevant to the plot.

```suggestion
    if (isCorrelatedScatter) {
      // compute correlation stats, but do it after the graph render so it doesn't delay
      // the visualizations or impact logging
      const spearmanRho = new SpearmanRho(scatter.data.x, scatter.data.y)
      spearmanRho.calc().then(value => setSpearman(value))
    }

    logScatterPlot(
      { scatter, genes, width: dimensions.width, height: dimensions.height },
      perfTimes
    )
```

Including this segment now lets us still analyze just front-end performance of Plotly rendering (via `perfTime:frontend:plot`, which isn't affected by this swap), while _also_ capturing this segment in `perfTime:frontend:other`, `perfTime:frontend`, and `perfTime:full`.  Including this segment now also avoids a discontinuity in those metrics for gene correlation plots in the likely case we want a named segment to track time to compute correlation.

Reply:
Sorry, the code comment is misleading.  The stats calculation is done asynchronously, so the change you suggest wouldn't impact logging except trivially.  I'm not sure I like waiting to log the plot until after the stats resolve, since a lot can happen, and I don't want to lose logging in the event the stats process errors or the user navigates away before they render.  I'll update the comment, and would suggest that we hold off on instrumenting the performance of these stats until we get a better sense of their usage and whether we'll be adding others.  (FWIW, the computations take ~1sec for 1MM cells in the console)

Reply:
Ah, right, it'd need to be e.g.:

```
if (isCorrelatedScatter) {
  // compute correlation stats, but do it after the graph render so it doesn't delay
  // the visualizations or impact logging
  const spearmanRho = new SpearmanRho(scatter.data.x, scatter.data.y)
  spearmanRho.calc().then(value => {
    setSpearman(value)
    logScatterPlot({ scatter, genes, width: dimensions.width, height: dimensions.height }, perfTimes)
  })
} else {
    logScatterPlot({ scatter, genes, width: dimensions.width, height: dimensions.height }, perfTimes)
  }
}
```

The benefits strike me as greater than the possible costs you describe, but I don't consider it blocking.  I appreciate the rationale and point-measurement data!